### ISSUE TITLE
Question: I can not wrap my head around logout test

### ISSUE THREAD

**Issue by Kyslik** [apiato/apiato#347]
https://github.com/apiato/apiato/blob/master/app/Containers/Authentication/UI/API/Tests/Functional/ApiLogoutTest.php

How on earth does that work?

**Comment by johannesschobel** [comment]
Hey @Kyslik ..
Maybe I can help you with this one:

The tests just tries to directly "logout" a given token - which was set to the database, and, therefore, calls the `DELETE /logout` route.. So if you want to understand this test, you basically need to understand the actual "controller-method" behind the route it calls.

The controller-method, however, "extracts" the `token` from the Request (obviously this here is a request, because the Test calls the API) and passes it to the `ApiLogoutAction`.

The Action itself extracts given information from the token and directly deletes this token from the database. That is all..

Regarding the `jwt.io: invalid signature` thing: In order to fully check the signature you would need to provide the public/private keys to the platform. As you may not have done that, the signature could not be validated - therefore, you should get the `Invalid Signature` message.
However, i would not suggest to submit public/private keys to any platform ;)

Cheers

**Comment by Kyslik** [comment]
Ok, so how does this test "deletes" token from database if there isn't any?!

>**Note**: we should use "updates token record to be **revoked**" instead of deletes

What I don't seem to understand is that: I am sending not properly signed token to the endpoint (while testing) and authentication is skipped, even though [logout route](https://github.com/apiato/apiato/blob/master/app/Containers/Authentication/UI/API/Routes/Logout.v1.public.php#L20-L22) has auth middleware applied.

**Comment by Mahmoudz** [comment]
@Kyslik you are right, as you can see from the tests, I am not creating any token to delete it (just for not spending much time on this). Since we're using the Passport package code which is already tested, I only want this test is to confirm that the request is passing in the pipeline and reaching passport, without caring about the result or the content. The token used is a fake one, I changed few characters in it, thus it's normal to return Invalid signature. I'd appreciate if someone will rewrite that test and make it more realistic (create token, authenticate the user and then logout).

**Comment by Kyslik** [comment]
@Mahmoudz (kindly) The question is still not-answered, why does this test does not throw up with **401 Unauthorized**?

**Comment by Mahmoudz** [comment]
@Kyslik I just checked the ApiLogoutAction, and I saw the problem you are talking about `DB::table('oauth_access_tokens')->where('id', '=', $id)->update(['revoked' => true]);` this tries to update the token to revoked weather it was found or not, and then the function returns true anyway. Kindly feel free to submit a PR fixing this by handling the cases where oauth_access_tokens not found or incorrect.. Hope I answered you this time. Let met know what do you think. Best,

**Comment by Kyslik** [comment]
The question is (updated since https://github.com/apiato/apiato/issues/347#issuecomment-352274618): 

> Why does the request get to pass through the controller when [**logout route**](https://github.com/apiato/apiato/blob/master/app/Containers/Authentication/UI/API/Routes/Logout.v1.public.php#L20-L22) is protected by **auth:api**. 

How does testing environment "disable" token signature checking.

**Comment by Mahmoudz** [comment]
mmm good question, that needs investigation.

**Comment by Kyslik** [comment]
I just made a test within my app:

 1. took the token
    - >`Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6ImUxODk3NjU3YTVlZWQxMDkyMThlZjVmOThlMGMzMzg5OTQ3MzNkZDYyZWM0Nzk5Y2E0NWQ1NjUyYmQ0NjFiZWUzZWQ5YjU1ZjNmMGE2NTA3In0.eyJhdWQiOiIyIiwianRpIjoiZTE4OTc2NTdhNWVlZDEwOTIxOGVmNWY5OGUwYzMzODk5NDczM2RkNjJlYzQ3OTljYTQ1ZDU2NTJiZDQ2MWJlZTNlZDliNTVmM2YwYTY1MDciLCJpYXQiOjE1MTI4NzgzNjMsIm5iZiI6MTUxMjg3ODM2MywiZXhwIjoxODI4MjM4MzYyLCJzdWIiOiI0Iiwic2NvcGVzIjpbXX0.fakesignaure`
 2. modified [FindRolesRequest](https://github.com/apiato/apiato/blob/master/app/Containers/Authorization/UI/API/Requests/FindRoleRequest.php#L21-L22) to not have any `roles` and `permissions` set (commented out)
 3. made a call to get all roles (using the token above) which results in a response with one role

> **Note**: when doing this with Postman I get expected response: 
> ```
>"status": "error",
>"code": 0,
>"message": "Unauthenticated.",
>"errors": "Missing or invalid Access Token!",
>"status_code": 401,
> ```
